server/src/main/java/org/elasticsearch/search/SearchModule.java

```diff
@@ -352,7 +371,12 @@ public Map<String, Highlighter> getHighlighters() {
     }
 
     private ValuesSourceRegistry registerAggregations(List<SearchPlugin> plugins) {
-        ValuesSourceRegistry.Builder builder = new ValuesSourceRegistry.Builder();
+        ValuesSourceRegistry.Builder builder = new ValuesSourceRegistry.Builder(
+            telemetryProvider == null
```

Comment:
telemetry provider shouldn't be null. Have you seen cases where it is?

Reply:
There's like 200 places in the tests where we create a `SearchModule`, as well as a couple places in production code where we do this: 

```
NamedWriteableRegistry registry = new NamedWriteableRegistry(new SearchModule(Settings.EMPTY, List.of()).getNamedWriteables());
```

(which seems like a bit of an anti-pattern, but also massively out of scope for this PR. 

At any rate, I defaulted it to null for those cases via a defaulting constructor (see line 315).  If there's something better than Null to default it to, I'm happy to do that instead, but I didn't see an obvious choice.

Reply:
There is `TelemetryProvider.NOOP`

Reply:
++
If you can use TelemetryProvider.NOOP, you could enforce it's always non-null and remove the null checks

Reply:
oh, spiff.  Not sure how I missed that.  I'll use that instead.